platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/ReadableRosterStore.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.service;
+
+import com.hedera.hapi.node.state.roster.Roster;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Provides read-only methods for interacting with Rosters.
+ */
+public interface ReadableRosterStore {
```

Comment:
In both comments, you haven't actually stated what makes it not good. What exactly is the pattern you are referring? If you say something isn't good, it will be helpful to know why.

>For platform state, there is a PlatformStateAccessor interface, and then an implementation each for a readable and writable version.
Similar to platform state, the roster state has its readable/writable stores through which its states are accessed. 

Are you saying `ReadableRosterStore` should implement `PlatformStateAccessor`? or `RosterStateAccessor`? If so, why?

Reply:
I'm saying a couple things. First, I'm expressing something similar to what @poulok [stated](https://github.com/hashgraph/hedera-services/pull/15442#discussion_r1763589931), that this is a poor class structure. 

Second, to suggest a solution to this problem, I'm saying that a much better pattern was used for the `PlatformState`. You're essentially doing the same thing here, so there would need to be a compelling reason to *not* follow the same pattern as used by `PlatformState`. You should create a `RosterStateAccessor`, and follow the same patterns as the `PlatformState` family of classes.

Reply:
I would agree with keeping and maintaining a consistent pattern / style so that people can recognize and reason about the same patterns.

Reply:
> I'm saying a couple things. First, I'm expressing something similar to what @poulok https://github.com/hashgraph/hedera-services/pull/15442#discussion_r1763589931, that this is a poor class structure.

@poulok's point is about removing a modification _definition_ from a `ReadableRosterStore` altogether instead of throwing an exception when called. I didn't take it away that the structure itself is fundamentally poor, but rather an improvement. There's a difference.

For context, currently, the "poor" class structure has `ReadableRosterStore` and `WritableRosterStore` implementations, where `WritableRosterStore` extends `ReadableRosterStore` exactly as it is for the platform state. 

Is what you're proposing that a `RosterStateAccessor` be introduced to front this "poor" structure? I ask because I think maybe if the structure is that poor, we probably want to steer clear altogether, except it isn't?

I think what you're proposing is equally just as "poor". As you can see, the ticket @poulok linked is proposing redoing the work to untie `PlatformStateAccessor` to both `ReadablePlatformStateStore` **and** `WritablePlatformStateStore`. The effect of that tying together is that `ReadablePlatformStateStore` had to implement the same set of APIs as `WritablePlatformStateStore` (i.e. write operations that throws exception when called on the readOnly interface), which is **exactly** what would happen if I just implemented a `RosterStateAccessor` as per your proposal.

@poulok I do like the proposal in @imalygin's ticket as it's more complete, thanks for sharing it. I'll clarify the details   today.

Reply:
@imalygin [Your PR](https://github.com/hashgraph/hedera-services/pull/15544) to modify the structure of platform state is relevant here. Probably worth coordinating to choose a consistent strategy, since this will likely be used for future state structures as well

Reply:
>@poulok I do like the proposal in @imalygin's ticket as it's more complete, thanks for sharing it. I'll clarify the details today.

This has been implemented and pushed in this review.